### PR TITLE
Better settings menu + basic debugging features

### PR DESCRIPTION
**Pull request by memedmusic**
Just redesigned the configure saladbind interface + added basic debugging features such as: clear cache, clear settings, remove miners.
Capture3
Capture1
Capture

### CODE REVIEW

**Review by Vukkyy — COMMENTED** [review]
is there any particular reason you have made all the debug options orange? looks kind of ugly and gives me cli+ vibes lol

**Comment by memedmusic** [comment]
> is there any particular reason you have made all the debug options orange? looks kind of ugly and gives me cli+ vibes lol

Nah, no reason lol. Would you prefer a different colour?

**Comment by Vukkyy** [comment]
> > is there any particular reason you have made all the debug options orange? looks kind of ugly and gives me cli+ vibes lol
> 
> Nah, no reason lol. Would you prefer a different colour?

how abut not setting one? white is... good enough, y'know

File: index.js [hunk 1]
```diff
@@ -64,7 +68,10 @@ if (process.argv[2]) {
 			console.log(chalk.bold.green(`SaladBind v${packageJson.version}`))
 			if (!fs.existsSync('./data/config.json')) {
 				console.log("Looks like this is your first time using SaladBind!\nLet's set it up. :)\n");
-				require("./setup").run(false);
+				require("./setup").first(false, false, true);
+			} else if (config.version == undefined) {
+				console.log("Looks like something isn't quite right with your settings. Lets fix them :)\n");
```
Comment (Vukkyy) [inline index.js:73]:
```suggestion
				console.log("Looks like something isn't quite right with your settings. Let's fix them :)\n");
```

File: setup.js [hunk 2]
```diff
@@ -154,25 +165,174 @@ async function continueSetup(clear) {
...
-			presence.mainmenu()
-			require("./index").menu();
-		}, 5000);
+		minerDetailsMethod = "Manual - Enter rig id"
+	}
+
+}
+
+
+
```
Comment (Vukkyy) [inline setup.js:174]: why do you like new lines so much lol

**Review by Vukkyy — CHANGES_REQUESTED** [review]

**Review by vtheskeleton — CHANGES_REQUESTED** [review]
After entering worker id the program exits
Also the orange looks kinda ugly, we should try to keep things white and only make notes orange (like in the miner selection screen where compatibility is noted in orange)

**Comment by vtheskeleton** [comment]
Also if it fails to find your worker id from logs it just goes to the menu anyway (even though there is a notice about it), still kinda bad design as the user may not realize something is wrong!

**Comment by memedmusic** [comment]
okei, I'll go change these thingies

File: setup.js (hunk 2 above)
Comment (memedmusic) [reply setup.js:174]: heh

**Comment by memedmusic** [comment]
Made colours better, removed a few extra blank lines, fixed spelling errors, fixed auto logs bug
Capture
Capture2

**Review by Vukkyy — CHANGES_REQUESTED** [review]
why does settings-main use `message:` in inquirer as its heading while the others console.log("settings - stuff") and do `message: choose an action`?

(i prefer the latter)

File: setup.js [hunk 3]
```diff
@@ -88,28 +108,22 @@ async function continueSetup(clear) {
 			if (rigID) rigID = rigID.split(": ")[1];
 			return rigID;
 		}
-		let rigID = getIDFromLogs("main.log") ?? getIDFromLogs("main.old.log")
```
Comment (TheBlueBurger) [inline setup.js:91]: Why did you remove let

**Review by TheBlueBurger — CHANGES_REQUESTED** [review]

File: setup.js (hunk 3 above)
Comment (memedmusic) [reply setup.js:91]: The save function is seperate from the minerDetails function, and using let doesnt expose the rigID variable to the whole script while removing let does

**Comment by memedmusic** [comment]
> why does settings-main use `message:` in inquirer as its heading while the others `console.log("settings - stuff")` and do `message: choose an action`?
> 
> (i prefer the latter, though it would be nice to not have the Settings - prefix, as... well, it already says Configure SaladBind, of course I'm in the settings lol)

Idrk lmao, I did it because, well.. i did. I guess I'll remove the console.log headers, remove the settings prefix, and change the 'message:' headers around

**Comment by memedmusic** [comment]
Changed the headers around a bit

File: setup.js [hunk 4]
```diff
@@ -248,11 +246,10 @@ async function first() {
 async function debugMenu() {
 	console.clear();
 	console.log(chalk.bold.cyan(`Configure SaladBind`))
-	console.log(chalk.bold.white("Settings - Debugging Tools"))
 	const promptResult = await inquirer.prompt([{
 		type: 'list',
 		name: "debug",
-		message: chalk.bold.white("Choose an action below"),
+		message: chalk.bold.white("Debugging tools"),
```
Comment (Vukkyy) [inline setup.js:252]: i said the latter, not the former... :p

**Review by Vukkyy — CHANGES_REQUESTED** [review]

**Comment by vtheskeleton** [comment]
Request a new review when you fix all the issues

File: setup.js (hunk 4 above)
Comment (memedmusic) [reply setup.js:252]: oh, right. mb

**Comment by Vukkyy** [comment]
unfortunately this seems to be heavily broken at the point of no return when i tried it out, but i appreciate you working on it! :)
